src/api/types/TypeRegistry.js

```diff
@@ -52,10 +53,31 @@ define(['./Type'], function (Type) {
      * @memberof module:openmct.TypeRegistry#
      */
     TypeRegistry.prototype.addType = function (typeKey, typeDef) {
+        this.standardizeType(typeDef);
         this.types[typeKey] = new Type(typeDef);
     };
 
     /**
+     * Takes a typeDef, standardizes it, and logs warnings about unsupported
+     * usage.
+     * @private
+     */
+    TypeRegistry.prototype.standardizeType = function (typeDef) {
+        if (typeDef.label) {
+            console.warn(
+                'DEPRECIATION WARNING typeDef: ' + typeDef.label + '.  ' +
+                '`label` is depreciated in type definitions.  Please use ' +
+                '`name` instead.  This will cause errors in a future version ' +
+                'of Open MCT.  For more information, see ' +
+                'https://github.com/nasa/openmct/issues/1568');
+            if (!typeDef.name) {
+                typeDef.name = typeDef.label;
+            }
+            delete typeDef.label;
```

Comment:
If this was occurring inside a tight loop or had a negative performance impact, that could be done.  In this case, I'd stick with the semantic option.